### PR TITLE
Fixes: creating tables

### PR DESCRIPTION
**Pull request by jmunozz**
- add filters on low values
-  fix soclong table name
- remove sensitive columns for rssi
- change how exposition is calculated

### CODE REVIEW

File: create_database/main.py [hunk 1]
```diff
@@ -176,24 +184,26 @@ def create_substance_soclong_table(_settings):
...
-    final_df = pd.merge(total_cases, hlt, left_index=True, right_index=True)
+    hlt.reset_index(["effet_hlt"], inplace=True)
+    final_df = pd.merge(decla_eff, hlt, left_index=True, right_index=True)
     final_df["pourcentage_cas"] = final_df.apply(
-        lambda x: float(x.n_decla_eff_hlt * 100 / x.n_cas), axis=1, result_type="expand"
+        lambda x: float(x.n_decla_eff_hlt * 100 / x.n_decla_eff), axis=1, result_type="expand"
     )
+    final_df.reset_index(["soc_long"], inplace=True)
+    final_df.drop(["n_decla_eff_hlt", "n_decla_eff"], inplace=True, axis=1)
     db.create_table_from_df(final_df, _settings["to_sql"])
```
Comment (lrahal) [inline create_database/main.py:206]: Check `create_hlt_table` in dash_database/ordei for computation of pourcentage_cas field

**Review by lrahal — CHANGES_REQUESTED** [review]

File: create_database/main.py [hunk 2]
```diff
@@ -101,9 +101,11 @@ def create_spe_patients_age_table(_settings):
 def create_substance_ordei_table(_settings):
     df = helpers.load_csv_to_df(_settings[0])
     df_by_years = df.groupby(["code", "annee"]).agg(conso_annee=("conso", "sum"), cas_annee=("cas", "sum"))
-    df_by_years["exposition"] = df_by_years["conso_annee"].apply(helpers.get_exposition_level, type="substance")
-    df_by_code = df_by_years.groupby("code").agg(conso=("conso_annee", "sum"), cas=("cas_annee", "sum"))
+    df_by_years["exposition_annee"] = df_by_years["conso_annee"].apply(helpers.get_exposition_level, type="substance")
```
[the remark this answers is not in the document]
Comment (jmunozz) [reply create_database/main.py:103]: done

File: create_database/main.py [hunk 3]
```diff
@@ -101,9 +101,11 @@ def create_spe_patients_age_table(_settings):
...
+    df_by_code = df_by_years.groupby("code").agg(conso=("conso_annee", "sum"), cas=("cas_annee", "sum"), taux_exposition=("conso_annee", helpers.get_total_exposition_level))
```
[the remark this answers is not in the document]
Comment (jmunozz) [reply create_database/main.py:104]: removed exposition_annee

File: create_database/main.py [hunk 4]
```diff
@@ -101,9 +101,11 @@ def create_spe_patients_age_table(_settings):
...
     final_df = df_by_years.join(df_by_code, on=["code"])
+    final_df = helpers.filter_low_values(final_df, ["cas", "cas_annee"])
+    final_df["taux_cas"] = final_df.apply(axis=1, func=lambda x: x.cas * 100000 / x.conso if x.cas >= 10 else None)
     final_df.reset_index(inplace=True, level=["annee"])
     db.create_table_from_df(final_df, _settings[0]["to_sql"])
```
[the remark this answers is not in the document]
Comment (jmunozz) [reply create_database/main.py:109]: done

File: create_database/main.py [hunk 5]
```diff
@@ -119,6 +121,7 @@ def create_substance_patients_sexe_table(_settings):
     )
     final_df = pd.merge(conso, conso_pct, on=["code", "sexe"])
     final_df.reset_index(inplace=True, level=["sexe"])
+    final_df.drop(["conso"], inplace=True, axis=1)
     db.create_table_from_df(final_df, _settings[1]["to_sql"])
```
[the remark this answers is not in the document]
Comment (jmunozz) [reply create_database/main.py:124]: done

File: create_database/main.py [hunk 6]
```diff
@@ -131,30 +134,35 @@ def create_substance_patients_age_table(_settings):
...
     )
     final_df = pd.merge(cas, cas_pct, on=["code", "sexe"])
+    final_df.drop(["cas"], axis=1, inplace=True)
     final_df.reset_index(inplace=True, level=["sexe"])
     db.create_table_from_df(final_df, _settings[3]["to_sql"])
 
 def create_substance_cas_age_table(_settings):
     df = helpers.load_csv_to_df(_settings[0])
     cas = df.groupby(["code", "age"])["cas"].sum().rename("cas")
+    cas = helpers.filter_serie_on_low_values(cas)
```
[the remark this answers is not in the document]
Comment (jmunozz) [reply create_database/main.py:157]: done

File: create_database/main.py [hunk 7]
```diff
@@ -176,24 +184,26 @@ def create_substance_soclong_table(_settings):
     total_case = total_case_per_sex_and_age.groupby("code").agg({ "n_cas": "sum" })
     decla_eff = df.groupby(["code", "soc_long"]).agg({ "n_decla_eff": "sum"}).reset_index(level="soc_long")
     final_df = pd.merge(total_case, decla_eff, left_index=True, right_on=["code"])
+    final_df = helpers.filter_df_on_low_values(final_df, ["n_decla_eff", "n_cas"])
```
[the remark this answers is not in the document]
Comment (jmunozz) [reply create_database/main.py:186]: done

File: create_database/main.py [hunk 8]
```diff
@@ -176,24 +184,26 @@ def create_substance_soclong_table(_settings):
...
     final_df["pourcentage_cas"] = final_df.apply(
-        lambda x: float(x.n_decla_eff * 100 / x.n_cas), axis=1, result_type="expand"
+        lambda x: float(x.n_decla_eff * 100 / x.n_cas) if x.n_decla_eff is not None and x.n_cas is not None else None, axis=1, result_type="expand"
```
[the remark this answers is not in the document]
Comment (jmunozz) [reply create_database/main.py:188]: I have an error when I do that

File: create_database/main.py [hunk 9]
```diff
@@ -176,24 +184,26 @@ def create_substance_soclong_table(_settings):
...
     )
-    # final_df.drop(["n_cas"], inplace=True, axis=1)
+    final_df.drop(["n_cas"], inplace=True, axis=1)
     db.create_table_from_df(final_df, _settings["to_sql"])
```
[the remark this answers is not in the document]
Comment (jmunozz) [reply create_database/main.py:191]: done

File: create_database/main.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (lrahal) [reply create_database/main.py:188]: Weird, because I tried and I don't get an error :D

**Pull request closed without merging**